src/glmfit.jl

```diff
@@ -579,6 +589,30 @@ fit(::Type{M},
     l::Link=canonicallink(d); kwargs...) where {M<:AbstractGLM} =
         fit(M, float(X), float(y), d, l; kwargs...)
 
+function fit(::Type{M},
+             f::FormulaTerm,
+             data,
+             d::UnivariateDistribution,
+             l::Link=canonicallink(d);
+             offset::Union{AbstractVector, Nothing} = nothing,
+             wts::Union{AbstractVector, Nothing} = nothing,
+             dofit::Bool = true,
```

Comment:
if they're calling `fit` instead of a constructor, I kinda feel like we shouldn't add the complexity of `dofit` kwarg.

Reply:
Yeah that doesn't make a lot of sense, but it's supported currently so I'd say we'd better keep supporting it. I've added deprecations so that we can remove them at some point.